api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.partitions.Partition;
+
+public interface SupportsPartitions extends Table {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
+
+  /**
+   * List all partitions
+   *
+   * @return The list of partitions
+   */
+  Partition[] listPartitions();
+
+  /**
+   * Get a partition by partition name
+   *
+   * @param partitionName the name of the partition
+   * @return the partition
+   * @throws NoSuchPartitionException if the partition does not exist
+   */
+  Partition getPartition(String partitionName) throws NoSuchPartitionException;
+
+  /**
+   * Check if a partition exists.
+   *
+   * @param partitionName The name of the partition.
+   * @return True if the partition exists, false otherwise.
+   */
+  default boolean partitionExists(String partitionName) {
+    try {
+      getPartition(partitionName);
+      return true;
+    } catch (NoSuchPartitionException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Add a partition with specified name and properties to the table.
+   *
+   * @param partition The partition to add.
+   * @return The created partition.
+   * @throws PartitionAlreadyExistsException If the partition already exists.
+   */
+  Partition addPartition(Partition partition) throws PartitionAlreadyExistsException;
```

Comment:
Is it necessary to return a `Partition` for `addPartition`?

Reply:
After adding a partition, certain properties such as `created time` and `location` may be automatically generated. Therefore, I propose returning a newly created partition to the user.

BTW, [HMS](https://github.com/apache/hive/blob/564d7e54d2360488611da39d0e5f027a2d574fc1/standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift#L2699-L2700) also return a partition for `addPartition`

Reply:
I see.